### ISSUE TITLE
Error in the CLIEngine.getErrorResults(report.results) returned array of objects.

### ISSUE THREAD

**Issue by Xflofoxx** [eslint/eslint#6155]
**What version of ESLint are you using?**
eslint@2.9.0
**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint@6.0.4

**What did you do? Please include the actual source code causing the issue.**
I'm using the grunt-eslint module that do this call:

```javascript
var results = report.results;
if (opts.quiet) {
	results = eslint.CLIEngine.getErrorResults(results);
}
var output = formatter(results);
```
**What did you expect to happen?**
I think it should filter the results getting only the "errors".

**What actually happened? Please include the actual, raw output from ESLint.**
The problem is that the function getErrorResults(results) only returns the filePath: and messages key, while the original report.results passed to the function contains also errorCount and warningCount keys.

Am I missing something?
Thanks.

![eslint-html-formatter-error](https://cloud.githubusercontent.com/assets/548893/15248499/d6e78604-191a-11e6-8b1e-0a82f901fdfb.JPG)

**Comment by alberto** [comment]
Confirmed using --quiet breaks the output of some formatters. 

Working on this.
